src/selection-util.js

```diff
@@ -23,24 +23,42 @@ export const isRowSelected = (rowIndex, selection) => {
   return rowIndex <= y2 && rowIndex >= y1
 }
 
-export const left = (selection, expand) => {
+export const left = (selection, expand, colCount, rowCount) => {
   const { x1, y1, x2, y2 } = selection
 
   if (!expand) {
-    const x = Math.max(x1 - 1, 0)
-    return { x1: x, x2: x, y1, y2: y1 }
+    let x = x1 - 1
+    let y = y1
+    if (x < 0) {
+      x = colCount - 1
+      if (y > 0) {
```

Comment:
i think there might be a bug here where the `y < 0` case is only taken care of when `x < 0`.

Reply:
on left underflow, need to move selection in only up direction, so when underflow occur, just need to check for y>0